workspaces/ballerina/ballerina-side-panel/src/components/editors/MultiModeExpressionEditor/ChipExpressionEditor/ChipExpressionBaseComponent.tsx

```diff
@@ -86,14 +87,27 @@ export const ChipExpressionBaseComponent = (props: ChipExpressionBaseComponentPr
     );
 
     const fetchUpdatedFilteredTokens = useCallback(async (value: string): Promise<number[]> => {
+        setIsLoading(true);
         const response = await expressionEditorRpcManager?.getExpressionTokens(
             value,
             props.fileName,
             props.targetLineRange.startLine
         );
+        setIsLoading(false);
         return response || [];
     }, [expressionEditorRpcManager]);
 
+    const getFnSignature = useCallback(async (value: string, cursorPosition: number) => {
+        if (props.extractArgsFromFunction) {
+            setIsLoading(true);
+            const fnSignature = await props.extractArgsFromFunction(value, cursorPosition);
```

Comment:
Shall we update extractArgsFromFunction return to exact type instead of `any` promise